chombot-common/src/data_watcher.rs

```diff
@@ -9,37 +9,44 @@ use tokio::time::sleep;
 
 const DATA_UPDATE_INTERVAL: Duration = Duration::from_secs(60 * 10);
 
-pub trait WatchableData<T> {
+pub trait WatchableData {
+    type Diff;
+
     #[must_use]
-    fn should_notify<'a>(&'a self, new: &'a Self) -> Option<(&'a T, &'a T)>;
+    fn should_notify<'a>(&'a self, new: &'a Self) -> Option<Self::Diff>;
     fn update(&mut self, new: Self);
```

Comment:
ideally `update` should also accept `Diff` but our usecases do not justify it I guess